docs/source/reference/northstar/serversiderui.rst

```diff
@@ -0,0 +1,187 @@
+
+**Example:**
+
+.. code-block:: javascript
+
+    void function CheckResponseToDummyPoll(entity player)
+    {
+        if(NSGetPlayerResponse(player) != -1)
+            print("Player has voted!")
+    }
+
+Large Message
+^^^^^
+
+Sends a large message to ``player`` which will appear in the top right corner.
+
+.. figure:: /_static/serversiderui/largemessage.gif
+  :align: center
+  :class: screenshot
+
+  Player POV
+  
+**Definition:**
+
+.. cpp:function:: void NSSendLargeMessageToPlayer( entity player, string title, string description, float duration, string image )
+
+**Example:**
+
+.. code-block:: javascript
+
+    void function SendDummyLargeMessage(entity player)
+    {
+        NSSendLargeMessageToPlayer(player,"I'm not a dummy >:(", "You are", 10, "ui/fd_tutorial_tip.rpak")
+    }
+
+Info Message
+^^^^^
+
+Sends a smaller message to ``player`` which will appear from the center right.
+
+.. figure:: /_static/serversiderui/info.gif
+  :align: center
+  :class: screenshot
+
+  Player POV
+
+**Definition:**
+
+.. cpp:function:: void NSSendInfoMessageToPlayer( entity player, string text )
+
+**Example:**
+
+.. code-block:: javascript
+  
+    void function SendDummyInfoMessage(entity player)
+    {
+        NSSendInfoMessageToPlayer(player, "If you're reading this your a dummy")
+    }
+
+PopUp
+^^^^^
+
+Send a small popup to ``player`` which will appear in the lower half of their screen under their cursor.
+
+.. figure:: /_static/serversiderui/popup.gif
+  :align: center
+  :class: screenshot
+
+  Player POV
+
+**Definition:**
+
+.. cpp:function:: void function NSSendPopUpMessageToPlayer( entity player, string text )
+
+**Example:**
+
+.. code-block:: javascript
+
+    void funcions SendDummyPopUp(entity player)
+    {
+        NSSendPopUpMessageToPlayer(player, "Dummy")
+    }
+
+Announcement
+^^^^^
+
+Sends a large announcement to ``player``.
+
+.. figure:: /_static/serversiderui/announcement.gif
+  :align: center
+  :class: screenshot
+
+  Player POV
+
+**Definition:**
+
+.. cpp:function:: void NSSendAnnouncementMessageToPlayer( entity player, string title, string description, vector color, int priority, int style )
+
+**Example:**
+
+.. code-block:: javascript
+
+      void function SendDummyAnnouncement(entity player)
+      {
+          NSSendAnnouncementMessageToPlayer(player, "Large dummy", "Small dummy", <1,1,0>, 1, ANNOUNCEMENT_STYLE_QUICK)
+      }
+
+Status
+^^^^^
+
+Status messages allow you to show live data to the player.
+Currently status messages are limited to 4 and there's no way to know if the player can see your message.
+
+.. figure:: /_static/serversiderui/status.gif
+   :align: center
+   :class: screenshot
+
+   Player POV
+
+**Definitions:**
+
+.. cpp:function:: void  NSCreateStatusMessageOnPlayer( entity player, string title, string description, string id )
+
+Creates a status message on ``player``. ``id`` is used to identify and edit the message, make sure your id is unique!
```

Comment:
We should mention `UniqueString()` more often based on the fact that I already saw usage of non-unique/deterministic strings for RUI message IDs.

```suggestion
Creates a status message on ``player``. ``id`` is used to identify and edit the message, make sure your id is unique!
To generate a unique id, use UniqueString().
```

There's also `UniqueString("Prefix_")` which returns a string which starts with "`Prefix_`" + random characters which is unique on the whole vm.

Reply:
Iirc `UniqueString()` doesn't exist, only `UniqueString( string )` does

Reply:
I think there's only `UniqueString( string )` and all it does is add a number to the string that gets larger every time you call the function or something like that

Reply:
also the string you get back isn't unique on the vm it's just unique from any output you got back from the function